waspc/data/Generator/templates/server/src/auth/utils.ts

```diff
@@ -34,28 +32,88 @@ export const authConfig = {
   successRedirectPath: "{= successRedirectPath =}",
 }
 
-export async function findUserBy(where: Prisma.{= userEntityUpper =}WhereUniqueInput): Promise<{= userEntityUpper =}> {
-  return prisma.{= userEntityLower =}.findUnique({ where });
+export async function findAuthIdentity(providerName: string, providerUserId: string) {
+  return prisma.{= authIdentityEntityLower =}.findUnique({
+    where: {
+      providerName_providerUserId: {
+        providerName,
+        providerUserId: providerUserId.toLowerCase(),
+      }
+    }
+  });
+}
+
+export async function findAuthIdentityByAuthId(authId: string) {
```

Comment:
Thanks to your comment I believe this method is kinda wrong. What if we had multiple identities for the same `Auth`? Then this wouldn't work! We should operate on `AuthIdentity` in this case and use the `Auth` only as a bridge towards the `User`. 

I'll refactor this piece a bit to concern itself with an _exact_ `AuthIdentity` and not the _first_ one (if there are multiple of them, this wouldn't be good).

Reply:
Ha good! I thought it was on purpose, you fetching the first Auth Identity you can find. And yeah, I think we should be writing the code assuming there can be multiple auth identities per Auth id, even if we don't yet have a way to have those in runtime (no merging).

Any thoughts on the `Auth` name? Too much to work to change it now I guess? And if you wanted to change it, would you have any ideas?

Reply:
We can change the `Auth` name, it wouldn't cause that much problems.

I thought `Auth` sounded generic enough since it's a connection to the "auth" relm from the business logic user to the various auth identities. 

I guess, `Account` and `Identities` could be a better set of names? But `Auth` and `AuthIdentity` also feels okay to me.

Reply:
Ok, if it feels good to you let's leave as it is. Feel free to resolve this comment once you read it.

Reply:
Okay, if there are no strong objections, let's keep it like this 👍